tests/integration/test_strict_config.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -eou pipefail
+
+set +e
```

Comment:
Doing `set +e` is quite error prone, for example right now you're allowing `bazelisk clean` to fail and it'll still run `bazelisk build` (maybe even successfully)

Reply:
Moved `clean` out.